lib/matplotlib/tests/test_legend.py

```diff
@@ -1457,47 +1456,26 @@ def test_boxplot_legend_labels():
     # Test that legend entries are generated from boxplots. `label` is the new
     # parameter for setting legend labels.
     np.random.seed(19680801)
-    data = np.random.lognormal(size=(68, 4), mean=1.5, sigma=0.2)
-    labels = list('ABCD')
-    colors = ['peachpuff', 'orange', 'tomato', 'plum']
-
-    legend_labels = ['peaches', 'oranges', 'tomatoes', 'plums']
-    fig, axs = plt.subplots(nrows=2,
-                            ncols=3,
-                            figsize=(10, 7),
-                            sharey=True)
-    # This plot will not generate a legend.
-    first = axs[0, 0].boxplot(data, patch_artist=True)
-    with pytest.warns(UserWarning):
-        axs[0, 0].legend()
-    empty_list = axs[0, 0].get_legend_handles_labels()
-    assert not empty_list[0] and not empty_list[1]
+    data = np.random.random((10, 4))
+    fig, axs = plt.subplots(nrows=1, ncols=3)
 
     # Testing legend labels and patch passed to legend.
-    second = axs[0, 1].boxplot(data, patch_artist=True, label=legend_labels)
-    for i, v in enumerate(second['boxes']):
-        assert v.get_label() == legend_labels[i]
+    bp1 = axs[0].boxplot(data, patch_artist=True, label=['A', 'B', 'C', 'D'])
+    for i, v in enumerate(bp1['boxes']):
+        assert v.get_label() == ['A', 'B', 'C', 'D'][i]
+    handles, labels = axs[0].get_legend_handles_labels()
+    assert labels == ['A', 'B', 'C', 'D']
+    assert all(isinstance(h, mpl.patches.PathPatch) for h in handles)
 
     # Testing legend without `box`.
-    third = axs[0, 2].boxplot(data, label=legend_labels, showbox=False)
+    bp2 = axs[1].boxplot(data, label=['A', 'B', 'C', 'D'], showbox=False)
     # Without a box, The legend entries should be passed from the medians.
-    for i, v in enumerate(third['medians']):
-        assert v.get_label() == legend_labels[i]
-
-    # compute the boxplot stats
-    stats = cbook.boxplot_stats(data, labels=labels, bootstrap=10000)
-
-    # The next 2 plots are called with `Axes.bxp`
-    fourth = axs[1, 0].bxp(stats, patch_artist=True, label=legend_labels)
-    for i, v in enumerate(fourth['boxes']):
-        assert v.get_label() == legend_labels[i]
-
-    fifth = axs[1, 1].bxp(stats, showbox=False, label=legend_labels)
-    for i, v in enumerate(fifth['medians']):
-        assert v.get_label() == legend_labels[i]
+    for i, v in enumerate(bp2['medians']):
+        assert v.get_label() == ['A', 'B', 'C', 'D'][i]
+    handles, labels = axs[1].get_legend_handles_labels()
+    assert labels == ['A', 'B', 'C', 'D']
+    assert all(isinstance(h, mpl.lines.Line2D) for h in handles)
```

Comment:
Since we're testing the labels, I also added a test that the handle is getting the right object. (patch for box and line for medians.)